pkg/securelaunch/helpers.go

```diff
@@ -0,0 +1,291 @@
+	return target, nil
+}
+
+/*
+ * NOTE: Caller's responsbility to unmount this..use return var mountPath to unmount in caller.
+ *
+ * obtains absolute file path based on input entered by user in policy file.
+ * inputVal is of format <block device identifier>:<path>
+ * inputVal e.g sda:/boot/foo.go , 4qccd342-12zr-4e99-9ze7-1234cb1234c4:/bar/zyx.go
+ * rw_option = true --> mount is of type read-write, else read-only.
+ *
+ * This function
+ * parses user input.
+ * mounts device
+ * Get absolute file path on mounted device.
+ * does NOT Unmount device, use returned mountPath to unmount...
+ * returns filePath, mountPath, error in that order
+ */
+func GetMountedFilePath(inputVal string, rw_option bool) (string, string, error) {
+	s := strings.Split(inputVal, ":")
+	if len(s) != 2 {
+		return "", "", fmt.Errorf("%s: Usage: <block device identifier>:<path>", inputVal)
+	}
+
+	// s[0] can be sda or UUID. if UUID, then we need to find its name
+	var deviceId string = s[0]
+	if !strings.HasPrefix(deviceId, "sd") {
+		if e := getBlkInfo(); e != nil {
+			return "", "", fmt.Errorf("GetMountedFilePath: getBlkInfo err=%s", e)
+		}
+		devices := storage.PartitionsByFsUUID(storageBlkDevices, s[0]) // []BlockDev
+		for _, device := range devices {
+			Debug("device =%s with fsuuid=%s", device.Name, s[0])
+			deviceId = device.Name
+		}
+	}
+
+	devicePath := filepath.Join("/dev", deviceId) // assumes deviceId is sda, devicePath=/dev/sda
+	Debug("Attempting to mount %s", devicePath)
+	var dev *diskboot.Device
+	var err error
+	if rw_option {
+		dev, err = diskboot.FindDeviceRW(devicePath) // FindDevice fn mounts , w rw option, devicePath=/dev/sda.
+	} else {
+		dev, err = diskboot.FindDevice(devicePath) // FindDevice fn mounts devicePath=/dev/sda.
+	}
+	if err != nil {
+		return "", "", fmt.Errorf("Mount %v failed, err=%v", devicePath, err)
+	}
+
+	Debug("Mounted %s", devicePath)
+	filePath := dev.MountPath + s[1] // mountPath=/tmp/path/to/target/file if /dev/sda mounted on /tmp
+	return filePath, dev.MountPath, nil
+}
+
+/*
+ * ScanKernelCmdLine() ([]byte, error)
+ * format sl_policy=<block device identifier>:<path>
+ * e.g sda:/boot/securelaunch.policy
+ * e.g 4qccd342-12zr-4e99-9ze7-1234cb1234c4:/foo/securelaunch.policy
+ */
+func ScanKernelCmdLine() ([]byte, error) {
+
+	Debug("ScanKernelCmdLine: scanning kernel cmd line for *sl_policy* flag")
+	val, ok := cmdline.Flag("sl_policy")
+	if !ok {
+		log.Printf("ScanKernelCmdLine: sl_policy cmdline flag is not set")
+		return nil, errors.New("Flag Not Set")
+	}
+
+	// val is of type sda:path
+	mntFilePath, mountPath, e := GetMountedFilePath(val, false) // false means readonly mount
+	if e != nil {
+		log.Printf("ScanKernelCmdLine: GetMountedFilePath err=%v", e)
+		return nil, fmt.Errorf("ScanKernelCmdLine: GetMountedFilePath err=%v", e)
+	}
+	Debug("ScanKernelCmdLine: Reading file=%s", mntFilePath)
+
+	d, err := ioutil.ReadFile(mntFilePath)
```

Comment:
check error?

Reply:
see line 204..We unmount first before checking error so that device is available sooner for other tasks..

Reply:
I see. Makes sense! This is kind of prompting me to improve the mounter interface. How do you feel about #1472?